deployment/hasura/migrations/Aerie/25_workspaces_setup/up.sql

```diff
@@ -0,0 +1,224 @@
+    on update cascade
+    on delete cascade
+);
+
+comment on table sequencing.workspace_collaborators is e''
+  'A collection of users who collaborate on the workspace alongside the workspace''s owner.';
+comment on column sequencing.workspace_collaborators.workspace_id is e''
+  'The plan the user is a collaborator on.';
+comment on column sequencing.workspace_collaborators.collaborator is e''
+  'The username of the collaborator';
+
+-- Update workspace table
+alter table sequencing.workspace
+ add column disk_location text,
+ add column parcel_id integer,
+ add foreign key (parcel_id)
+    references sequencing.parcel
+    on update cascade
+    on delete restrict;
+
+comment on column sequencing.workspace.id is e''
+  'The unique id of the workspace.';
+comment on column sequencing.workspace.disk_location is e''
+  'The location of the workspace on disk.';
+comment on column sequencing.workspace.parcel_id is e''
+  'The parcel that files in the workspace use.';
+comment on column sequencing.workspace.updated_at is e''
+  'Time the workspace was last updated.';
+
+-- Data migration: parcel_id
+-- Create a new copy of the workspace for every parcel after the first.
+
+/*
+  Big SQL statement that,
+   - Updates existing workspaces to have a parcel id
+  If there are multiple parcels in use across the user sequences in the workspace, it:
+    1) creates a clone workspace for each additional parcel, with the name <OLD WS NAME>(<PARCEL_NAME>). I.E. "My Workspace (2.8.0-Lite)"
+    2) moves the user sequences that are using the additional parcels to the correct clone ws
+ */
+with parcels_per_ws as (
+  -- Number each parcel in use on each workspace
+  select parcel_id, workspace_id, pname, row_number() over (partition by workspace_id) as wrow
+    from (select distinct parcel_id, workspace_id, p.name as pname
+  from sequencing.user_sequence us join sequencing.parcel p on us.parcel_id = p.id
+  order by workspace_id, parcel_id) a
+), inserted_ws(new_wid, name, pid) as (
+  -- Create a new workspace for each additional parcel used on the workspace
+  insert into sequencing.workspace (name, parcel_id, owner, created_at, updated_by, updated_at)
+   select name || ' (' || ppw.pname ||')' as name, ppw.parcel_id, owner, created_at, updated_by, updated_at
+   from parcels_per_ws ppw join sequencing.workspace ws on (ppw.workspace_id = ws.id)
+   where ppw.wrow > 1
+   returning id, name, parcel_id
+), update_ws as (
+  -- Update the already existing workspaces to have parcel ids
+  update sequencing.workspace ws
+    set parcel_id = ppw.parcel_id
+    from parcels_per_ws ppw
+    where ws.id = ppw.workspace_id and wrow = 1
+)
+  -- Move user sequences to the new workspaces as needed
+  update sequencing.user_sequence us
+    set workspace_id = iws.new_wid
+  from inserted_ws iws, parcels_per_ws ppw, sequencing.workspace ws
+  where us.workspace_id = ws.id
+    and iws.name = ws.name || ' (' || ppw.pname ||')'
```

Comment:
minor: @Mythicaeda can you clarify if this line is needed or not? Was trying to understand if it was safe to remove & just match on `id`s instead

Reply:
nevermind, we understand why this is needed, but would prefer to keep the old workspace ID in `inserted_ws` & then do a match like `and iws.old_workspace_id = ws.id` here to avoid any string/name matching brittleness